integration-cli/docker_api_containers_test.go

```diff
@@ -724,13 +725,21 @@ func (s *DockerSuite) TestContainerAPIVerifyHeader(c *check.C) {
 	// Try with no content-type
 	res, body, err := create("")
 	c.Assert(err, checker.IsNil)
-	c.Assert(res.StatusCode, checker.Equals, http.StatusBadRequest)
+	if versions.LessThan(testEnv.DaemonAPIVersion(), "1.32") {
+		c.Assert(res.StatusCode, checker.Equals, http.StatusInternalServerError)
```

Comment:
Wonder if we should actually backport the fix for this status; 500 may be due to a bug / not properly handled condition

Reply:
@AntaresS agree that we should look at what the old test was testing, I doubt we were testing for 500.

Reply:
Hm, right, it's a bit tricky, because it's effectively not related to the API version, but to the version of the daemon that's running (i.e., the "internal server error" was caused by an unhandled error condition, so defaulted to a 500). i.e. using API version 1.31 (or older) on a _current_ version of the daemon would still have the fix, so will return `http.StatusBadRequest`.

Was thinking if we should do a `statuscode != 200` or `status code >= 400`, but then we'd miss possible regressions if a `500` is returned on the current codebase.

Reply:
I tend to agree with @thaJeztah… if it's not related to the API version we shouldn't check the API version.. We could also check the `dockerd` version number (or the version field in the `info` endpoint for a cleaner way of doing the check) and act based on that ?

Reply:
@vdemeester @thaJeztah how about we add a TODO explaining the problem. I agree with both of you, there's just not a great way of doing order comparisons between dockerd versions. We can find a proper fix in a followup?

Reply:
@AntaresS ok let's fix this by only asserting the correct error, if API ≥ 1.32.

Reply:
@thaJeztah @tiborvass OK here is what I did. Let's check for only API ≥ 1.32 for now. I added TODO for future reference.
```
if versions.GreaterThanOrEqualTo(testEnv.DaemonAPIVersion(), "1.32") {
      c.Assert(res.StatusCode, checker.Equals, http.StatusBadRequest)
}
```